src/Illuminate/Filesystem/Filesystem.php

```diff
@@ -578,11 +578,10 @@ public function glob($pattern, $flags = 0)
     /**
      * Get an array of all files in a directory.
      *
-     * @param  string  $directory
      * @param  bool  $hidden
      * @return \Symfony\Component\Finder\SplFileInfo[]
      */
-    public function files($directory, $hidden = false, array|string|int $depth = 0)
+    public function files(string|array $directory, $hidden = false, array|string|int $depth = 0)
```

Comment:
That is not entirely correct. The docblock _is_ useful, when typed as `array` as well, as we can then narrow the `array` type further down to what is actually allowed in it as Symfony also does: https://github.com/symfony/symfony/blob/8a58816cc8fd78732d878e20c975dc64e09dca86/src/Symfony/Component/Finder/Finder.php#L629

```suggestion
    *
    * @param  string|string[]  $directory
    * @param  bool  $hidden
    * @return \Symfony\Component\Finder\SplFileInfo[]
    */
    public function files(string|array $directory, $hidden = false, array|string|int $depth = 0)
```